mysql-test/suite/binlog/t/binlog_mysqlbinlog_warn_stop_datetime.test

```diff
@@ -1,42 +1,86 @@
+--let $binlog_f1= query_get_value(SHOW MASTER STATUS, File, 1)
+create table t1 (a int);
+insert into t1 values (1);
+--eval SET TIMESTAMP= UNIX_TIMESTAMP('$b1_timestamp2')
+insert into t1 values (2);
+--eval SET TIMESTAMP= UNIX_TIMESTAMP('$b1_timestamp3')
+flush binary logs;
+--let $binlog_f2= query_get_value(SHOW MASTER STATUS, File, 1)
+--eval SET TIMESTAMP= UNIX_TIMESTAMP('$b2_timestamp1')
+insert into t1 values (3);
+insert into t1 values (4);
+--eval SET TIMESTAMP= UNIX_TIMESTAMP('$b2_timestamp2')
+--let $binlog_f2_gtid_after_midpoint= `SELECT @@GLOBAL.gtid_binlog_pos`
+insert into t1 values (5);
+insert into t1 values (6);
+insert into t1 values (7);
+--let $binlog_f2_last_gtid= `SELECT @@GLOBAL.gtid_binlog_pos`
+--eval SET TIMESTAMP=UNIX_TIMESTAMP('$b2_timestamp3')
+flush binary logs;
+drop table t1;
 
---echo Case: Stop datetime between records, must not see warning.
---echo # MYSQL_BINLOG --short-form --stop-datetime='2001-09-08 21:46:55' MYSQLD_DATADIR/master-bin.000001 --result-file=ignored_output_file
---exec $MYSQL_BINLOG --short-form --stop-datetime='2001-09-08 21:46:55' $MYSQLD_DATADIR/master-bin.000001 --result-file=$ignored_output_file 2>&1
+--echo # Ensuring binary log order is correct
+--let $binlog_f1_show= query_get_value(SHOW BINARY LOGS, Log_name, 1)
+if (`SELECT strcmp('$binlog_f1','$binlog_f1_show') != 0`)
+{
+  --echo # Real binlog_f1: $binlog_f1
+  --echo # First binlog in SHOW BINLOG FILES: $binlog_f1_show
+  --die Wrong order of binary log files in SHOW BINARY LOGS
+}
+--let $binlog_f2_show= query_get_value(SHOW BINARY LOGS, Log_name, 2)
+if (`SELECT strcmp('$binlog_f2','$binlog_f2_show') != 0`)
+{
+  --echo # Real binlog_f2: $binlog_f2
+  --echo # First binlog in SHOW BINLOG FILES: $binlog_f2_show
+  --die Wrong order of binary log files in SHOW BINARY LOGS
+}
 
---echo Case: Stop datetime at EOF, must not see warning.
---echo # MYSQL_BINLOG --short-form --stop-datetime='2001-09-08 21:46:55' MYSQLD_DATADIR/master-bin.000001 --result-file=ignored_output_file
---exec $MYSQL_BINLOG --short-form --stop-datetime='2001-09-08 21:46:55' $MYSQLD_DATADIR/master-bin.000001 --result-file=$ignored_output_file 2>&1
+--echo #
+--echo #
+--echo # Test using --read-from-remote-server
+--echo #
+--let $use_local_files= 0
+--source binlog_mysqlbinlog_warn_stop_datetime.inc
 
---echo Case: Stop datetime after EOF, must see warning.
---echo # MYSQL_BINLOG --short-form --stop-datetime='2035-01-19 03:14:05' MYSQLD_DATADIR/master-bin.000001 --result-file=ignored_output_file
---exec $MYSQL_BINLOG --short-form --stop-datetime='2035-01-19 03:14:05' $MYSQLD_DATADIR/master-bin.000001 --result-file=$ignored_output_file 2>&1
-
-DROP TABLE t1;
-
---remove_file $ignored_output_file
+--echo #
+--echo #
+--echo # Test using local binlog files
+--echo #
+--let $use_local_files= 1
+--source binlog_mysqlbinlog_warn_stop_datetime.inc
```

Comment:
Personal preference: the variable should be `$read_from_remote_server` so that `use_local_files` is default.